TINY-10268: manage `list` inside an `li` surrounded by 2 not `li` elements (#9086)
tinymce/tinymce: modules/tinymce/CHANGELOG.md

- Search and replace plugin would incorrectly find matching text inside non-editable root elements. #TINY-10162
- Search and replace plugin would incorrectly find matching text inside SVG elements. #TINY-10162
- Removed use of `async` for editor rendering which caused visual blinking when reloading the editor in-place. #TINY-10249

> - A `list` inside a `li` surrounded by 2 not `li` elements was just considered a `content` of that `li`. #TINY-10268

Reviewer: ```suggestion
- List items containing both text and a nested list caused some list operations to fail. #TINY-10268
```

I'm not sure if "fail" is the correct word, but it's close enough I guess?

Author: @TheSpyder since we are managing a nested list between 2 text, could it be better something like this?
```
- List items containing a nested list between two text caused some list operations to fail. #TINY-10268
```

Reviewer: Well that was my point, the bug isn't just for between two text items. But if that's all you're going to address in this ticket that seems fine - just seems like a waste to make it a whole new ticket when the fix looks like only one or two lines of code different to what's in this PR.

Author: changed based to this comment: https://github.com/tinymce/tinymce/pull/9086#discussion_r1367003088

- Search and replace plugin would incorrectly find matching text inside non-editable root elements. #TINY-10162
- Search and replace plugin would incorrectly find matching text inside SVG elements. #TINY-10162
- Removed use of `async` for editor rendering which caused visual blinking when reloading the editor in-place. #TINY-10249

> - List items containing the first and the last element as text and a nested list caused some list operations to fail. #TINY-10268

Reviewer 2: Maybe not sure if its better:
```suggestion
- List items containing a list element surrounded by non list nodes would cause some list operations to fail. #TINY-10268
```
Writing changelog items is harder than naming things. :smile:

Author: changed
